pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -261,3 +262,8 @@ func makeLabels(s *v1alpha1.TaskRun) map[string]string {
 	labels[pipeline.GroupName+pipeline.TaskRunLabelKey] = s.Name
 	return labels
 }
+
+// TrimContainerNamePrefix trim the container name prefix to get the corresponding step name
+func TrimContainerNamePrefix(containerName string) string {
+	return strings.TrimPrefix(containerName, containerPrefix)
```

Comment:
are you removing the prefix "build-step" but keeping the random string in the end of the name?

Reply:
@pivotal-nader-ziada indeed, the random string in the end is still there, we should trim it too

Reply:
@pivotal-nader-ziada @vdemeester yes, this method only trims the `build-step` prefix, for this case, the method is used to handle the build step name, the build step name consists of `build-step` prefix and the task step name, the task step name is defined by user, there should be no random string in it unless user defined it, right?

Reply:
@skeeey the pod actually get a random suffix at the end of the name — and are limited in size. I think we should actually use the pod labels to get the task name (see below an extract of a describe on a pod)

```
Name:               build-push-run-pod-04cde6
Namespace:          default
Priority:           0
PriorityClassName:  <none>
Node:               minikube/192.168.122.229
Start Time:         Tue, 02 Apr 2019 12:17:01 +0200
Labels:             tekton.dev/task=build-push-kaniko
                    tekton.dev/taskRun=build-push-run
```

Reply:
@vdemeester I'm confused :-), why we need a `task` name？in this case, we want to add the step name to each step that is in the `taskrun` status, and in fact, each `task` step corresponds to a container in a pod,  we can get the container name from `ContainerStatus` directly, and the container name consists of `build-step` prefix and the `task` step name, I think we can trim the `build-step` prefix to get the actual step name, is my understand right?

Reply:
@skeeey lol sorry, I messed up :joy: You're right :wink:

Reply:
@skeeey right so we just need to trim the last 7 characters of the pod name then (to get a clean step name :angel: )